satori/resources/resource_user_custom_attributes.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	mergedAttributes, err := mergeUserAndConfiguredAttributesMap((*result).CustomAttributes, d)
+	if err != nil {
+		return nil, err
+	}
+
+	rawMergedAttributes, err := json.Marshal(mergedAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	err = d.Set(Attributes, string(rawMergedAttributes))
+	if err != nil {
+		return nil, err
+	}
+	d.SetId((*result).UserId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	attr := api.UserWithCustomAttributes{}
+
+	attr.UserId = d.Get(UserId).(string)
+	attr.CustomAttributes = mergedAttributes
+
+	return &attr, err
+}
+
+// This function merges the attributes from the
+func mergeUserAndConfiguredAttributesMap(userAttrMap map[string]interface{}, d *schema.ResourceData) (map[string]interface{}, error) {
+	changeMap := make(map[string]interface{})
+	currAttributesMap := make(map[string]interface{})
+	rawAttributes := d.Get(Attributes).(string)
+
+	fileContent, readFileErr := os.ReadFile(rawAttributes)
+
+	if readFileErr == nil {
+		rawAttributes = string(fileContent)
+	}
+
+	err := json.Unmarshal([]byte(rawAttributes), &currAttributesMap)
+	if err != nil {
+		return nil, err
+	}
+
+	for key := range currAttributesMap {
+		// Keeping change map updated in-case there is a diff between currAttributesMap and userAttrMap
+		// which means that the attributes should be updated.
+		userVal, _ := userAttrMap[key]
+		changeMap[key] = userVal
+
+		// Delete the key, so it will not be duplicated down the road.
+		delete(userAttrMap, key)
+	}
+
+	// If there are any attributes left in the attributes set.
+	// Add them to the changeMap
+	for key, userValue := range userAttrMap {
+		changeMap[key] = userValue
+	}
+
+	return changeMap, nil
+}
+
+func resourceUserCustomAttributesUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	c := m.(*api.Client)
+
+	result, err := updateSatoriAttributes(d, c)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	fmt.Println(result)
+
+	return diags
+}
+
+func updateSatoriAttributes(d *schema.ResourceData, c *api.Client) (*api.UserWithCustomAttributes, error) {
+	input, err := resourceToUserAttribute(d)
+
+	if err != nil {
+		return nil, err
+	}
+	result, err := c.UpdateUserCustomAttributes(input)
+	return result, err
+}
+
+func resourceUserCustomAttributesDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	c := m.(*api.Client)
+
+	result, err := deleteUserSatoriAttributes(d, c)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	fmt.Println(result)
+
+	return diags
+}
+
+func deleteUserSatoriAttributes(d *schema.ResourceData, c *api.Client) (*api.UserWithCustomAttributes, error) {
+	input := api.UserWithCustomAttributes{}
+	input.UserId = d.Get(UserId).(string)
+
+	result, err := c.DeleteUserCustomAttributes(&input)
+
+	// Setting terraform-id from the empty string ---> triggers delete
+	d.SetId("")
```

Comment:
Why do you need this? didn't understand

Reply:
d.SetId("") triggers the terraform's object deletion from the terraform.state